datafusion/core/src/physical_optimizer/join_selection.rs

```diff
@@ -1289,27 +1302,59 @@ mod tests_statistical {
         );
     }
 
+    #[rstest(
+        join_type, projection, small_on_right,
+        case::inner(JoinType::Inner, vec![1], true),
+        case::left(JoinType::Left, vec![1], true),
+        case::right(JoinType::Right, vec![1], true),
+        case::full(JoinType::Full, vec![1], true),
+        case::left_anti(JoinType::LeftAnti, vec![0], false),
```

Comment:
I think it will be better to add a swap test for LeftAnti & LeftSemi?

Reply:
Since small_on_right is false for  LeftAnti & LeftSemi,  the swap rule don't work for those tests. But it doesn't matters.